packages/react-scripts/scripts/utils/createJestConfig.js

```diff
@@ -14,14 +14,17 @@ const paths = require('../../config/paths');
 module.exports = (resolve, rootDir, isEjecting) => {
   // Use this instead of `paths.testsSetup` to avoid putting
   // an absolute filename into configuration after ejecting.
+  const setupTestsMatches = paths.testsSetup.match(/src\/setupTests\.(.+)/);
+  const setupTestsFileExtension =
+    (setupTestsMatches && setupTestsMatches[1]) || 'js';
   const setupTestsFile = fs.existsSync(paths.testsSetup)
-    ? '<rootDir>/src/setupTests.js'
+    ? `<rootDir>/src/setupTests.${setupTestsFileExtension}`
```

Comment:
@brunolemos  from this comment https://github.com/facebook/create-react-app/pull/4837#issuecomment-428834087

So does it work if I use TypeScript? I removed it https://github.com/straw-hat-team/jest-config/commit/9cfcc0024c3ffbbe1350161aae9d9a8bf5424534 because you told me it wouldn't work 😄

Reply:
Both `setupFiles` and `setupTestFrameworkScriptFile` are passed through transformers

Reply:
Yes it works, setupProxy is the one that doesn't